core/src/main/java/tc/oc/pgm/regions/ResizedRegion.java

```diff
@@ -0,0 +1,63 @@
+package tc.oc.pgm.regions;
+
+import org.bukkit.util.Vector;
+import tc.oc.pgm.api.region.Region;
+import tc.oc.pgm.util.math.TransformMatrix;
+
+public class ResizedRegion extends TransformedRegion {
+  private final Vector min, max;
+  private final boolean relative;
+  private TransformMatrix matrix;
+  private TransformMatrix inverse;
+
+  public ResizedRegion(Region region, Vector min, Vector max, boolean relative) {
+    super(region);
+    this.min = min;
+    this.max = max;
+    this.relative = relative;
+  }
+
+  @Override
+  protected Vector transform(Vector point) {
+    if (matrix == null) ensureInitialized();
+    return matrix.transform(point);
+  }
+
+  @Override
+  protected Vector untransform(Vector point) {
+    if (inverse == null) ensureInitialized();
+    return inverse.transform(point);
+  }
+
+  @Override
+  public Bounds getBounds() {
+    ensureInitialized();
+    return super.getBounds();
+  }
+
+  private void ensureInitialized() {
+    if (matrix != null) return;
+
+    var oldBounds = region.getBounds();
+    var oldSize = oldBounds.getSize();
+
+    if (relative) {
+      min.multiply(oldSize);
+      max.multiply(oldSize);
+    }
+
+    this.bounds =
+        new Bounds(oldBounds.getMin().subtract(min), oldBounds.getMax().add(max));
+    var newSize = bounds.getSize();
+
+    this.matrix = TransformMatrix.concat(
+        TransformMatrix.untranslate(oldBounds.getMin()),
+        TransformMatrix.scale(newSize.clone().divide(oldSize)),
```

Comment:
Should we be doing a check here for zero? 
 suppose the possible presence of infinity or -infinity could be intended Although NaN could pose issues

Reply:
Yeah for empty regions it probs makes sense to just leave these matrices as identities as its irrelevant